### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by Schwusch**
Two new services emerge out of this change, Neptune and DocDB. These were placed [under RDS](https://pub.dev/documentation/aws_rds_api/latest/neptune-2014-10-31/Neptune-class.html) before due to sharing the same endpoint prefix. They seem to be vastly different services though.

 - Readme and example is created for the latest version of the APIs.
 - Switch candidate priority in packageBaseName to differentiate
   APIs sharing endpointPrefix.
 - Update Member model class with new fields.

### CODE REVIEW

**Comment by Schwusch** [comment]
This fixes the crash mentioned in https://github.com/agilord/aws_client/issues/144, but I ran this test:
```dart
import 'package:aws_sqs_api/sqs-2012-11-05.dart';
import 'package:http/http.dart';
import 'package:http/testing.dart';
import 'package:test/test.dart';

void main() {
  test('foo', () async {
    final client = MockClient((request) async {
      print(request.body);
      return Response('<root><SendMessageResult><foo></foo></SendMessageResult></root>', 200);
    });

    final sqs = SQS(
      region: 'foo',
      credentials: AwsClientCredentials(accessKey: 'null', secretKey: 'null'),
      client: client,
    );
    await sqs.sendMessage(
        messageBody: '{"foo":6}',
        queueUrl: 'http://foo.com',
        messageGroupId: 'bar',
        messageDeduplicationId: '666',
        messageAttributes: {
          'companyId': MessageAttributeValue(
              dataType: 'string', stringValue: 'companyId')
        });
  });
}
```
The request body of this is:
`Action=SendMessage&MessageAttributes.entry.1.key=companyId&MessageAttributes.entry.1.value.entry.1.key=DataType&MessageAttributes.entry.1.value.entry.1.value=string&MessageAttributes.entry.1.value.entry.2.key=StringValue&MessageAttributes.entry.1.value.entry.2.value=companyId&MessageBody=%7B%22test%22%3A+%22cool%22%7D&MessageDeduplicationId=4a7f57c07cf611eaf1cde9f670f7ada1&MessageGroupId=bar&QueueUrl={sqsUrl}&Version=2012-11-05`

Which differs a bit from what @lucasdina says is a valid:
`Action=SendMessage&MessageAttribute.1.Name=testattrib&MessageAttribute.1.Value.DataType=String&MessageAttribute.1.Value.StringValue=test&MessageBody=%7B%22test%22%3A+%22cool%22%7D&MessageDeduplicationId=4a7f57c07cf611eaf1cde9f670f7ada1&MessageGroupId=AgriPayFarms&QueueUrl={sqsUrl}&Version=2012-11-05`

File: shared_aws_api/lib/src/protocol/query.dart [hunk 1]
```diff
@@ -119,6 +119,10 @@ Iterable<MapEntry<String, String>> _flatten(
     return;
   }
 
+  if (data is! Map) {
+    data = data.toJson();
```
Comment (isoos) [inline shared_aws_api/lib/src/protocol/query.dart:123]: I'm working on a patch that does not need this, maybe remove it for the time being?

File: generator/lib/model/api.dart [hunk 2]
```diff
@@ -79,9 +87,9 @@ class Api {
 
   String get packageBaseName {
     final candidates = <String>[
-      metadata.endpointPrefix,
       metadata.uid?.split('-20')?.first,
       metadata.className.toLowerCase(),
+      metadata.endpointPrefix
```
Comment (isoos) [inline generator/lib/model/api.dart:92]: Why this change? What's the impact?

File: shared_aws_api/lib/src/protocol/query.dart (hunk 1 above)
Comment (isoos) [reply shared_aws_api/lib/src/protocol/query.dart:123]: On a second thought this can be here, just add a // TODO: to remove it once we call `.toJson` at request building time.

File: generator/lib/model/api.dart (hunk 2 above)
Comment (Schwusch) [reply generator/lib/model/api.dart:92]: RDS, DocDB and Neptune share the same endpoint prefix, hence they're all int the [RDS library](https://pub.dev/documentation/aws_rds_api/latest/index.html).
This change means they will be landing in separate packages.
